tools/pull-request-labeler/src/utils/split-message-into-components.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CommitMessage, CommitType } from '../interfaces/commit-message';
+
+const expression = /(?<type>feat|fix|docs|style|refactor|perf|test|build|chore|barista)(?<component>\(.*?\))?:(?<message>.*)/gims;
```

Comment:
whats about ci – you could loop over the value of the enum and join them by the `|` and dynamical build this regex